src/gradata/enhancements/rule_canary.py

```
@@ -0,0 +1,301 @@
+
+        # Count corrections in this category since canary started
+        correction_count = 0
+        try:
+            corr_row = conn.execute(
+                "SELECT COUNT(*) as cnt FROM events WHERE type = 'CORRECTION' "
+                "AND data_json LIKE ? AND CAST(session AS INTEGER) >= ?",
+                (f'%"{rule_category}"%', start_session),
+            ).fetchone()
+            if corr_row:
+                correction_count = corr_row["cnt"]
+        except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)